--- GNUmakefile.in	2024-02-03 20:20:30.453900939 -0800
+++ GNUmakefile.in	2024-02-04 01:29:58.938823923 -0800
@@ -155,7 +155,7 @@
     $(eval $c-in-$1:; + if [ -f submodules/$1/Makefile ]; then $(MAKE) submodules/$1/Makefile && $(MAKE) -C submodules/$1 $c; fi)
     $(eval $c: $c-in-$1))
 
-git-checkout-in-$1:; cd @srcdir@/.. && git submodule update --init M2/submodules/$1
+git-checkout-in-$1:
 git-update-in-$1:; cd @srcdir@/submodules/$1 && git checkout master && git pull
 git-status-in-$1:; cd @srcdir@/submodules/$1 && git status
 git-clean-in-$1:; cd @srcdir@/submodules/$1 && git clean -Xdf
@@ -205,7 +205,7 @@
 	@ false
 
 @srcdir@/submodules/$1/configure @srcdir@/submodules/$1/Makefile.in: @srcdir@/submodules/$1/configure.ac @srcdir@/submodules/$1/Makefile.am
-	cd @srcdir@/submodules/$1 && NOCONFIGURE=1 ./autogen.sh
+	cd @srcdir@/submodules/$1 && NOCONFIGURE=1 sh ./autogen.sh
 endef
 $(foreach s,@SUBLIST@,$(eval $(call submodule-rules,$s)))
 $(foreach s,@BUILDSUBLIST@,$(eval $(call build-submodule-rules,$s)))
